### DISCUSSION TITLE
How to ignore property in UrlEncoded mode?

### DISCUSSION THREAD

**Discussion by CodingOctocat** [reactiveui/refit#1680] [Q&A] [answered]
How to ignore some properties in FooRequest?

```csharp
[Post(Foo.QueryPath)]
Task<string> QueryAsync(
    [Body(BodySerializationMethod.UrlEncoded)] FooRequest fooRequest,
    CancellationToken cancellationToken = default);

class FooRequest
{
    [AliasAs("aliasbar1")]
    public string Bar1 { get; set; }

    [Refit.IgnoreRequest] // like ???
    public string Bar2 { get; set; }
}
```

**Comment by glennawatson** [comment, marked as answer]
There is no [IgnoreRequest]-style attribute in Refit for UrlEncoded bodies. When you pass an object as [Body(BodySerializationMethod.UrlEncoded)], Refit (via FormValueMultimap) enumerates every public readable property and serializes it; the only property it skips automatically is one whose value is null.

So your options are:

- Make the getter non-public, so the form serializer does not see it. FormValueMultimap only includes properties with a public getter:

```csharp
class FooRequest
{
    [AliasAs("aliasbar1")]
    public string Bar1 { get; set; }

    public string Bar2 { internal get; set; }   // not serialized into the form
}
```

- Or do not put it on the body type at all - keep the request DTO limited to the fields you actually send, and carry Bar2 separately.

- Or leave it null when you do not want it sent (null values are skipped).

Note: [JsonIgnore] does not exclude a property from UrlEncoded form serialization - that path does not use the JSON serializer. The supported way to exclude a field is to keep it out of the public-getter surface of the body type.
